src/lib/y2network/interface.rb

```diff
@@ -22,8 +22,19 @@ class Interface
     # @return [String] Device name (eth0, wlan0, etc.)
     attr_reader :name
 
+    # Constructor
+    #
+    # @param name [String] Interface name (e.g., "eth0")
     def initialize(name)
       @name = name
     end
+
+    # Determines whether two interfaces are equal
+    def ==(other)
+      return false unless other.respond_to?(:name)
```

Comment:
hmm, I am not sure that this is expected. What about simple `other.is_a? Interface` ? This way it can be == for object that define name, which is not so uncommon and produce strange results. Or better what is intention of this lazy type check?